### PR TITLE
Add setFactory permission to GceDiscoveryPlugin

### LINKED ISSUE #16485: Plugin does not work with ES 2.2.0
Get complaints about java permissions but adding them doesn't seem to fix...

[2016-02-05 11:39:42,346][WARN ][discovery.gce ] [homer] Exception caught during discovery: access denied ("java.lang.RuntimePermission" "setFactory") java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "setFactory") at java.security.AccessControlContext.checkPermission(AccessControlContext.java:372) at java.security.AccessController.checkPermission(AccessController.java:559) at java.lang.SecurityManager.checkPermission(SecurityManager.java:549) at java.lang.SecurityManager.checkSetFactory(SecurityManager.java:1625) at javax.net.ssl.HttpsURLConnection.setSSLSocketFactory(HttpsURLConnection.java:362) at com.google.api.client.http.javanet.NetHttpTransport.buildRequest(NetHttpTransport.java:145) at com.google.api.client.http.javanet.NetHttpTransport.buildRequest(NetHttpTransport.java:62) at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:863) at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:419) at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352) at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469) at org.elasticsearch.cloud.gce.GceComputeServiceImpl$1$1.run(GceComputeServiceImpl.java:94) at org.elasticsearch.cloud.gce.GceComputeServiceImpl$1$1.run(GceComputeServiceImpl.java:90) at java.security.AccessController.doPrivileged(Native Method) 

Glad to provide any information needed to help troubleshoot.

I answered 'y' (yes) to the prompt to add security options to java, also attempted to add them manually. Still no luck. Uninstalled and reinstalled the plugin...

I rolled back to 2.1.1 and it works again...

### PR DESCRIPTION
**Pull request by s1monw**
This commit adds a missing permission and a simple test that
ensures we discover other nodes via a mock http endpoint.

Closes #16485 

@rjernst I had to make this integ test a ordinary testcase since it runs against an internal cluster. I didn't explore the test fixture path yet since it also requires a test plugin etc. I wonder if we can make this one test run against the internal cluster while others runs against external clusters? I am not super happy with the test but it's a major step forward since it really tests the integration of this plugins while everything else isn't really testing it. It also would have caught the security issue though

### CODE REVIEW

File: plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeService.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.function.Function;
```
Comment (dadoonet) [inline plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeService.java]: Why this import?

**Comment by dadoonet** [comment]
Interesting. I like the idea of using an HTTP server to serve the mock responses.
Thanks @s1monw for fixing this issue.

LGTM

**Comment by rjernst** [comment]
This looks ok to me. It is good to have some kind of test for GCE. I'm apprehensive about checking in the keyfile, but I guess it is just for a test...

At another time, I am happy to look at moving this to a fixture, as well as generating the keyfile, but this PR at least gets us into a better state than we are now.

**Comment by rmuir** [comment]
+1

**Comment by jasontedor** [comment]
I have the same apprehension as @rjernst but given 

> At another time, I am happy to look at moving this to a fixture, as well as generating the keyhole

I'm good with the change too. Thanks for picking this up @s1monw.

**Comment by s1monw** [comment]
@rjernst @jasontedor I pushed a new commit generating the keystore from gradle

**Comment by rjernst** [comment]
Looks good, thanks!

**Comment by jasontedor** [comment]
LGTM.

**Pull request merged**